engine.py

```diff
@@ -95,15 +95,33 @@ def post_app_init(self):
 
         from sgtk.platform.qt import QtCore
 
-        if not self.has_ui:
-            # no UI. everything after this requires the UI!
-            return
-        
         tk_houdini = self.import_module("tk_houdini")
         bootstrap = tk_houdini.bootstrap
+        
+        if not self.has_ui:
+            bootstrap.bootstrap(self.sgtk, self.context)
 
         if bootstrap.g_temp_env in os.environ:
 
+            # Figure out the tmp OP Library path for this session
+            oplibrary_path = os.environ[bootstrap.g_temp_env].replace("\\", "/")
+            
+            def _load_otls():
+                self._load_otls(oplibrary_path)
+
+            if not self.has_ui:
+                _load_otls()
+                # no UI. everything after this requires the UI!
+                return
+            else:
+                # We have the same problem here on Windows that we have above with
+                # the population of the shelf. If we defer the execution of the otl
+                # loading by an event loop cycle, Houdini loads up quickly.
+                if sys.platform.startswith("win"):
```

Comment:
Does this no longer need to happen if the UI is not present?
I guess the shelf won't be present, so maybe the issue goes away.

Reply:
ah sorry must not have been thinking clearly, obviously QtCore isn't going to be present. We'll have to test if the issue is still present and maybe find a different method here.

Reply:
I've tested this and it seems to be fine, on Windows with hython, so looks like this change should be fine.